### PR TITLE
shimgvw: Enable zooming

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Enable zooming in and zooming out at shimgvw.
JIRA issue: [CORE-12338](https://jira.reactos.org/browse/CORE-12338)

### CODE REVIEW

File: dll/win32/shimgvw/shimgvw.c [hunk 1]
```diff
@@ -41,6 +39,108 @@ WNDPROC PrevProc = NULL;
...
+    if (ZoomPercents >= MAX_ZOOM)
+        SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_ZOOMP, FALSE);
+    else
+        SendMessage(hToolBar, TB_ENABLEBUTTON, IDC_ZOOMP, TRUE);
+
+    /* redraw */
+    InvalidateRect(hDispWnd, NULL, TRUE);
+}
+
+static void ZoomOut(void)
```
Comment (HBelusca) [inline dll/win32/shimgvw/shimgvw.c]: These two functions can be merged into a single one, with a parameter indicating which "direction" (zooming in or out) to do

Comment (katahiromz) [reply dll/win32/shimgvw/shimgvw.c]: Done.

File: dll/win32/shimgvw/shimgvw.c [hunk 2]
```diff
@@ -710,7 +838,7 @@ ImageView_CreateWindow(HWND hwnd, LPWSTR szFileName)
     WndClass.style          = CS_HREDRAW | CS_VREDRAW;
     WndClass.hIcon          = LoadIcon(hInstance, MAKEINTRESOURCE(IDI_APPICON));
     WndClass.hCursor        = LoadCursor(hInstance, IDC_ARROW);
-    WndClass.hbrBackground  = (HBRUSH)COLOR_WINDOW;
+    WndClass.hbrBackground  = (HBRUSH)GetStockObject(NULL_BRUSH);   /* less flicker */
```
Comment (gedmurphy) [inline dll/win32/shimgvw/shimgvw.c]: Assuming you're repainting the entire HDC, setting this to NULL should eliminate the flicker. There's no reason to paint the background and then the foreground in the majority of cases.

**Comment by gedmurphy** [comment]
I'm happy with these changes. Unless anyone object, I'll merge this on sunday

File: dll/win32/shimgvw/shimgvw.c [hunk 3]
```diff
@@ -597,6 +713,12 @@ ImageView_WndProc(HWND hwnd, UINT Message, WPARAM wParam, LPARAM lParam)
         }
         break;
 
+        case WM_MOUSEWHEEL:
+            ImageView_OnMouseWheel(hwnd,
+                (SHORT)LOWORD(lParam), (SHORT)HIWORD(lParam),
+                (SHORT)HIWORD(wParam), (UINT)LOWORD(wParam));
```
Comment (HBelusca) [inline dll/win32/shimgvw/shimgvw.c]: I would strongly recommend here the usage of: `GET_X_LPARAM(lParam);` and `GET_Y_LPARAM(lParam);` for the X and Y deltas.

Comment (katahiromz) [reply dll/win32/shimgvw/shimgvw.c]: Done.

**Pull request merged**